tests/nvmeof/test_ceph_nvmeof_gateway_sub_scale.py

```diff
@@ -82,104 +89,192 @@ def configure_listeners(config, _cls, command, ceph_cluster):
+                )
+
+        LOG.info("Verification of QoS values is successful")
+
+    for num in range(1, namespaces_sub + 1):
+        qos_settings = [
+            {
+                "rw-ios-per-second": 1500,
+                "rw-megabytes-per-second": 6,
+                "r-megabytes-per-second": 3,
+                "w-megabytes-per-second": 3,
+            },
+            {
+                "rw-ios-per-second": 150,
+                "rw-megabytes-per-second": 59,
+                "r-megabytes-per-second": 59,
+                "w-megabytes-per-second": 59,
+            },
+        ]
+        for _, qos_values in enumerate(qos_settings):
+            subnqn = f"nqn.2016-06.io.spdk:cnode{sub_num}{f'.{group}' if group else ''}"
+            config = {"args": {"nsid": num, "subsystem": subnqn, **qos_values}}
+
+            apply_qos(config)
+            LOG.info(f"Applied QoS for NSID {num} on subsystem {subnqn}")
+
+            verify_qos(subnqn, qos_values, num)
+            LOG.info(f"Updated QoS for NSID {num} on subsystem {subnqn}")
 
 
 def configure_namespaces(
     config, _cls, command, ceph_cluster, node, init_config, rbd_obj, io_param
 ):
     subsystems = config["args"].pop("subsystems")
     namespaces = config["args"].pop("namespaces")
-    image_size = config["args"].pop("image_size")
+    image_size = config["args"].pop("image_size", None)
     group = config["args"].pop("group", None)
-    pool = config["args"].pop("pool")
+    pool = config["args"].pop("pool", None)
     namespaces_sub = int(namespaces / subsystems)
+    if command == "add":
+        for sub_num in range(1, subsystems + 1):
+            init_config.update(
+                {
+                    "subnqn": f"nqn.2016-06.io.spdk:cnode{sub_num}{f'.{group}' if group is not None else ''}"
+                }
+            )
+            initiators(ceph_cluster, node, init_config)
+            name = generate_unique_id(length=4)
+            LOG.info(sub_num)
+            LOG.info(subsystems)
+
+            def configure_namespace(num, config):
+                rbd_obj.create_image(pool, f"{name}-image{num}", image_size)
+                LOG.info(num)
+                config["args"].clear()
+                subnqn = f"nqn.2016-06.io.spdk:cnode{sub_num}{f'.{group}' if group is not None else ''}"
+                config = {
+                    "base_cmd_args": {"format": "json"},
+                    "args": {
+                        "rbd-image": f"{name}-image{num}",
+                        "rbd-pool": pool,
+                        "subsystem": subnqn,
+                        "force": True,
+                    },
+                }
+                namespace_func = fetch_method(_cls, command)
+                _, namespaces = namespace_func(**config)
+
+                _config = {
+                    "base_cmd_args": {"format": "json"},
+                    "args": {
+                        "subsystem": subnqn,
```

Comment:
Please include nsid here, we shall not list history of NS we had which is too huge for logs
With nsid, we can log only that NS through list